src/libcsg/modules/io/h5mdtrajectoryreader.h

```diff
@@ -180,6 +183,24 @@ class H5MDTrajectoryReader : public TrajectoryReader {
   //
   int vec_components_;
 
+  // Length scaling - VOTCA internally uses nm as the length unit.
+  bool unit_module_enabled_ = false;
+  double length_scaling_ = 1.0;
+  double velocity_scaling_ = 1.0;
+  double force_scaling_ = 1.0;
+
+  // Convert map from SI prefixes to nm (VOTCA base length unit).
+  std::unordered_map<std::string, double> length_units_votca_scaling_factors = std::unordered_map<std::string, double>(
```

Comment:
I think an unordered map for 5 element is a bit overkill, but I do not care in the end.